pkg/agent/nodeportlocal/k8s/npl_controller.go

```diff
@@ -598,11 +597,10 @@ func (c *NPLController) waitForRulesInitialization() {
 				continue
 			}
 			allNPLPorts = append(allNPLPorts, rules.PodNodePort{
-				NodePort:  npl.NodePort,
-				PodPort:   npl.PodPort,
-				PodIP:     pod.Status.PodIP,
-				Protocol:  npl.Protocol,
-				Protocols: npl.Protocols,
```

Comment:
We need to remove `Protocols` of `rules.PodNodePort` and stop using it in `AddAllRules`. Otherwise `AddAllRules` won't work, I'm curious why e2e tests didn't fail. https://github.com/antrea-io/antrea/pull/5917 can catch the issue but currently it has to expect wrong output.

Reply:
Removed.
The e2e codes are updated accordingly for the change it this PR and the `PodNodePort` is not used before in NPL e2e test cases.